### ISSUE TITLE
ld -lGL could not be found

### ISSUE THREAD

**Issue by ProRedHD** [go-flutter-desktop/go-flutter#286]
Hi,

I don't know if this is the right place to ask but I transfered my existing go-flutter-desktop project over to a new pc and installed go-flutter as descriped. I checked everything but it should work but I get this error. 

I already spent 2 hours trying to fix it and I am clueless.

OS: Kubuntu 19.04

```
hover: Using engine from cache
hover: Cleaning the build directory
hover: Bundling flutter app
hover: Failed to retrieve the organization name: open android/app/src/main/AndroidManifest.xml: no such file or directory
hover: Compiling 'go-flutter' and plugins
github.com/go-gl/glfw/v3.2/glfw
github.com/go-gl/gl/v3.3-core/gl
# github.com/go-gl/glfw/v3.2/glfw
/usr/bin/ld: -lGL kann nicht gefunden werden
collect2: error: ld returned 1 exit status
# github.com/go-gl/glfw/v3.2/glfw
In file included from ../../../go/pkg/mod/github.com/go-gl/glfw@v0.0.0-20190409004039-e6da0acd62b1/v3.2/glfw/c_glfw_linbsd.go:24:
../../../go/pkg/mod/github.com/go-gl/glfw@v0.0.0-20190409004039-e6da0acd62b1/v3.2/glfw/glfw/src/linux_joystick.c: In function ‘_glfwInitJoysticksLinux’:
../../../go/pkg/mod/github.com/go-gl/glfw@v0.0.0-20190409004039-e6da0acd62b1/v3.2/glfw/glfw/src/linux_joystick.c:224:42: warning: ‘%s’ directive output may be truncated writing up to 255 bytes into a region of size 9 [-Wformat-truncation=]
             snprintf(path, sizeof(path), "%s/%s", dirname, entry->d_name);
                                          ^~~~~~~
In file included from /usr/include/stdio.h:867,
                 from /usr/include/X11/Xcursor/Xcursor.h:26,
                 from ../../../go/pkg/mod/github.com/go-gl/glfw@v0.0.0-20190409004039-e6da0acd62b1/v3.2/glfw/glfw/src/x11_platform.h:39,
                 from ../../../go/pkg/mod/github.com/go-gl/glfw@v0.0.0-20190409004039-e6da0acd62b1/v3.2/glfw/glfw/src/internal.h:169,
                 from ../../../go/pkg/mod/github.com/go-gl/glfw@v0.0.0-20190409004039-e6da0acd62b1/v3.2/glfw/glfw/src/x11_init.c:28,
                 from ../../../go/pkg/mod/github.com/go-gl/glfw@v0.0.0-20190409004039-e6da0acd62b1/v3.2/glfw/c_glfw_linbsd.go:19:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:67:10: note: ‘__builtin___snprintf_chk’ output between 12 and 267 bytes into a destination of size 20
   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        __bos (__s), __fmt, __va_arg_pack ());
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# github.com/go-gl/gl/v3.3-core/gl
/usr/bin/ld: -lGL could not be found
collect2: error: ld returned 1 exit status
hover: Go build failed: exit status 2
```

**Comment by Drakirus** [comment]
That's the right place!
Your PC doesn't seems to have the OpenGL dev libraries https://en.wikibooks.org/wiki/OpenGL_Programming/Installation/Linux.
You can install them and it should work.

BUT before doing that, can you try `hover run --opengl=none` (and `hover build linux --opengl=none`).

>
 ```
$ hover run --help
Flags:
      --opengl string   The OpenGL version specified here is only relevant for external texture plugin (i.e. video_plugin).
                        If 'none' is provided, texture won't be supported. Note: the Flutter Engine still needs a OpenGL compatible context. (default "3.3")
```

**Comment by Drakirus** [comment]
Humm, I take back what I said, `--opengl=none` shouldn't fix your issue. Only installing OpenGL will.

**Comment by ProRedHD** [comment]
Thank you for the quick answer.

I'm afraid installing OpenGL didn't help. It says I got all the packages installed and the same error happens. Do you have any other suggestions?

**Comment by Drakirus** [comment]
What is your linux distribution?

**Comment by Drakirus** [comment]
What is the output of `$ ls /usr/lib/libGL*`?
My best be is: `sudo apt install --reinstall libgl1-mesa-glx`

Since I cannot reproduce this error, and I don't really know where it comes from:
-> https://stackoverflow.com/a/54161024
-> https://stackoverflow.com/a/54397235
-> https://stackoverflow.com/a/32184137
-> https://stackoverflow.com/a/33458318

I guess you have already tried them ;(

**Comment by Drakirus** [comment]
If you really can't get OpenGL to be installed on your pc, use cross-compiling.

Hover enables you to compile apps to other platforms than the one you are working on. 
(`hover build windows` will produce a .exe even if you are not working on windows). 

This cross-compiling feature is possible by Docker. And Docker cross-compiling is automatically trigger when your-OS != buildOS. 

If you want to use Docker compilation when your-OS  == buildOS, you need to provide the `--docker` flag.

TL;DR: `hover run - - docker`

**Comment by ProRedHD** [comment]
Thanks for your help in solving this problem! I really appreciate it. I reinstalled libgl1-mesa-glx and moved some files around and the application is starting up. 

However it is stuck at the "Loading..." stage and the console is telling me that there is no handler found for "flutter/isolate" and the sqlite-plugin

```
go-flutter: no handler found for channel flutter/isolate
go-flutter: no handler found for channel com.tekartik.sqflite

```

I am gonna see if installing android and running flutter itself once will fix. "pub get" doesn't help either.

**Comment by jld3103** [comment]
You need to add the sqflite plugin for go-flutter: https://github.com/nealwon/go-flutter-plugin-sqlite

**Comment by ProRedHD** [comment]
Right, thanks. I completely forgot that I did that last time. Stupid mistake! Everything is working as expected.
